x/staking/keeper/msg_server.go

```diff
@@ -184,8 +199,23 @@ func (k msgServer) EditValidator(ctx context.Context, msg *types.MsgEditValidato
 			return nil, errorsmod.Wrap(sdkerrors.ErrLogic, err.Error())
 		}
 
-		if msg.CommissionRate.LT(minCommissionRate) {
-			return nil, errorsmod.Wrapf(sdkerrors.ErrInvalidRequest, "commission rate cannot be less than the min commission rate %s", minCommissionRate.String())
+		maxCommissionRate, err := k.MaxCommissionRate(ctx)
+		if err != nil {
+			return nil, err
+		}
+
+		if maxCommissionRate.Equal(minCommissionRate) {
+			if !msg.CommissionRate.Equal(minCommissionRate) {
+				return nil, errorsmod.Wrapf(types.ErrCommissionLTRateNotEqual, "if the validity minimum and maximum commission rates are equal (%s), the commission rate (%s) must be equal too", msg.CommissionRate.String(), minCommissionRate.String())
+			}
+		} else {
+			if msg.CommissionRate.LT(minCommissionRate) {
+				return nil, errorsmod.Wrapf(types.ErrCommissionLTMinRate, "commission rate (%s) cannot be less than the min commission rate %s", msg.CommissionRate.String(), minCommissionRate.String())
+			}
+
+			if msg.CommissionRate.GT(maxCommissionRate) {
+				return nil, errorsmod.Wrapf(types.ErrCommissionLTMaxRate, "cannot set validator (%s) commission to more than minimum rate of %s", msg.CommissionRate.String(), maxCommissionRate.String())
+			}
```

Comment:
Duplicated code.

Could also be simplified with a single condition, something like this:
```go
if msg.CommissionRate.LT(minCommissionRate) || msg.CommissionRate.GT(maxCommissionRate) {
	return nil, errorsmod.Wrapf(types.ErrCommissionOutOfBound,
		"commission rate (%s) must be between %s and %s",
		msg.CommissionRate.String(), minCommissionRate.String(), maxCommissionRate.String(),
	)
}
```